components/ChallengeMaterial.tsx

```diff
@@ -102,6 +104,47 @@ export const ChallengeQuestionCard: React.FC<ChallengeQuestionCardProps> = ({
   const { data } = React.useContext(SessionContext)
   const username = _.get(data, 'userInfo.username', '')
   const diff = _.get(currentChallenge, 'submission.diff', false)
+  let files = null
+
+  if (diff) files = gitDiffParser.parse(diff)
+
+  const renderFile = ({ hunks, newPath }: File, i: number) => {
+    const oldValue: String[] = []
+    const newValue: String[] = []
+
+    hunks.forEach(hunk => {
+      hunk.changes.forEach(change => {
+        if (change.isDelete) oldValue.push(change.content)
+        else if (change.isInsert) newValue.push(change.content)
+        else {
+          oldValue.push(change.content)
+          newValue.push(change.content)
+        }
+      })
+    })
+
+    const syntaxHighlight = (str: string): any => {
+      if (!str) return
+
+      const language = Prism.highlight(
+        str,
+        Prism.languages.javascript,
+        'javascript'
+      )
+      return <span dangerouslySetInnerHTML={{ __html: language }} />
+    }
+
+    return (
+      <ReactDiffViewer
+        key={i}
```

Comment:
It is not a best recommended to use `index` as `key` with React.
Since we use lodash you might use `_.uniqueId()` [link](https://lodash.com/docs/4.17.15#uniqueId)

Reply:
updated :+1: